services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java

```diff
@@ -174,29 +174,47 @@ public static AuthorizationTokenService instance() {
     }
 
     private static void fireErrorEvent(EventBuilder event, String error, Exception cause) {
-        event.detail(Details.REASON, cause == null || cause.getMessage() == null ? "<unknown>" : cause.getMessage())
-                .error(error);
+        if (cause instanceof CorsErrorResponseException) {
+            // cast the exception to populate the event with a more descriptive reason
+            CorsErrorResponseException originalCause = (CorsErrorResponseException) cause;
+            event.detail(Details.REASON, originalCause.getErrorDescription() == null ? "<unknown>" : originalCause.getErrorDescription())
+                    .error(error);
+        } else {
+            event.detail(Details.REASON, cause == null || cause.getMessage() == null ? "<unknown>" : cause.getMessage())
+                    .error(error);
+        }
+
         logger.debug(event.getEvent().getType(), cause);
     }
-    
-    public Response authorize(KeycloakAuthorizationRequest request) {
+
+    public Response authorize(KeycloakAuthorizationRequest request, EventBuilder event) {
```

Comment:
This is not really necessary because the `request` will never be null. The check we have is just a defensive check. It is probably a code assertion where a null `request` is a bug and not expected at all.

Reply:
@jmrodriguez Could you please revert this change? See comment above.

Reply:
done

Reply:
@jmrodriguez I think the change is not reflected here?

Reply:
@pedroigor I cannot remove the EventBuilder event parameter as because fireErrorEvent function requires it as parameter, and we use that function multiple times in the authorize method.

Could you please clarify how this should be resolved?

Reply:
Sure. Basically, change the signature and include the first statement as follows:

```
public Response authorize(KeycloakAuthorizationRequest request) {
        EventBuilder event = request.getEvent();
        ...
}
```

You should be able to get the `EventBuilder` from the request.

Reply:
I understand now. I'm pushing the changes shortly